web/src/main/java/com/epam/esm/controllers/TagController.java

```diff
@@ -0,0 +1,123 @@
+package com.epam.esm.controllers;
+
+import com.epam.esm.entities.Certificate;
+import com.epam.esm.entities.Tag;
+import com.epam.esm.exceptions.ResponseException;
+import com.epam.esm.exceptions.service.ServiceException;
+import com.epam.esm.exceptions.validation.ValidationErrorMessage;
+import com.epam.esm.services.TagService;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+/**
+ * Provides REST API functionality for {@link Tag}
+ */
+@RestController
+@RequestMapping(value = "/tags", produces = {MediaType.APPLICATION_JSON_VALUE})
+public class TagController {
+
+    private static final String ERROR_MESSAGE = "response.idErrorMessage";
+    private static final String ERROR_DETAILS = "response.idErrorDetails";
+    private final TagService tagService;
+
+    @Autowired
+    public TagController(TagService tagService) {
+        this.tagService = tagService;
+
+    }
+
+    /**
+     * Method provide functionality for getting {@link Tag} collection.
+     * <p> GET /web/tags
+     *
+     * @return <ul>ResponseEntity containing {@link List<Certificate>} which contains all certificates in data source.
+     * Response status: 200 OK</ul>
+     */
+    @GetMapping
+    public ResponseEntity getAllTags(@RequestAttribute(name = "name", required = false) String name) {
+        ResponseEntity responseEntity;
```

Comment:
@KukharchukDaniil All logic to the service layer.

Reply:
fixed